### PR TITLE
Bazel FUSE: Add enkit commands

### PR DESCRIPTION
**Pull request by minor-fixes**
This change adds enkit subcommands for bazel artifacts mounting, as laid
out in
https://docs.google.com/document/d/1L9skX1BKY-AZ15GVypQWxiy360UkUo5mNXglC9vvO9g/edit#.

These commands will need to tie together functionality from faketree,
bb_clientd management, and zipfile/tempdir management, so unlike other
enkit subcommand trees there is no pre-existing logical source location
to put the commands themselves. For now, they are added as a package
under `enkit` itself.

Tested: Output for help of each command:

`enkit outputs mount`:

```
Mount the build outputs of a particular invocation

Usage:
  enkit outputs mount [invocation ID] [flags]

Examples:
  $ enkit outputs mount 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
        Mounts outputs from build 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07 to the
        default location.

Flags:
  -h, --help   help for mount

Global Flags:
      --help-all              Show all the flags available, even those that are less useful and normally hidden.
      --outputs_root string   Root dir of mounted outputs (default "/home/bminor/outputs")
```

`enkit outputs unmount`:

```
Unmount the build outputs of a particular invocation

Usage:
  enkit outputs unmount [invocation ID] [flags]

Aliases:
  unmount, umount

Examples:
  $ enkit outputs unmount 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
        Unmounts outputs from build 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07 from the
        default location.

Flags:
  -h, --help   help for unmount

Global Flags:
      --help-all              Show all the flags available, even those that are less useful and normally hidden.
      --outputs_root string   Root dir of mounted outputs (default "/home/bminor/outputs")
```

`enkit outputs run`:

```
Mount build artifacts and execute an optional command on them

Usage:
  enkit outputs run [flags]

Examples:
  $ enkit outputs run --invocation=73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
        Launches a shell with build outputs from a particular build re-rooted so that
        paths are correct within the outputs themselves.

  $ enkit outputs run --dir=/tmp/some_dir
        Launches a shell with build outputs in /tmp/some_dir re-rooted so that paths
        are correct within the outputs themselves.

  $ enkit outputs run --zip=/tmp/some.zip
        Unpacks the named zip into a tempdir, and then reroots the artifacts so that
        paths are correct within the outputs themselves.

  $ enkit outputs run --zip=/tmp/some-zip -- find .
        Runs "find ." in the unpacked zip after it is rerooted.

Flags:
      --dir_path string        If set, the path to already-unpacked artifacts to reroot
  -h, --help                   help for run
      --invocation_id string   If set, the build's invocation ID from which to mount artifacts
      --zip_path string        If set, the path to a zipped outputs file to unpack and re-root

Global Flags:
      --help-all              Show all the flags available, even those that are less useful and normally hidden.
      --outputs_root string   Root dir of mounted outputs (default "/home/bminor/outputs")
```

`enkit outputs shutdown`:

```
Unmount all builds under particular directory

Usage:
  enkit outputs shutdown [flags]

Examples:
  $ enkit outputs shutdown
        Unmounts all builds in the given output root and resets the output root to a pristine state.

Flags:
  -h, --help   help for shutdown

Global Flags:
      --help-all              Show all the flags available, even those that are less useful and normally hidden.
      --outputs_root string   Root dir of mounted outputs (default "/home/bminor/outputs")
```

Jira: INFRA-469
Jira: INFRA-488

### CODE REVIEW

File: enkit/outputs/commands.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	root *Root
+}
+
+func NewShutdown(root *Root) *Shutdown {
+	command := &Shutdown{
+		Command: &cobra.Command{
+			Use:   "shutdown",
+			Short: "Unmount all builds under particular directory",
+			Example: `  $ enkit outputs shutdown
+	Unmounts bb_clientd, and deletes all links to build artifacts.`,
```
Comment (ccontavalli) [inline enkit/outputs/commands.go:177]: remove references to bb_clientd, we are using bb_clientd now as implementation, but our average developer may not know / will likely not have direct visibility into it?

File: enkit/outputs/commands.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+type Shutdown struct {
+	*cobra.Command
+	root *Root
+}
+
+func NewShutdown(root *Root) *Shutdown {
+	command := &Shutdown{
+		Command: &cobra.Command{
+			Use:   "shutdown",
```
Comment (ccontavalli) [inline enkit/outputs/commands.go:174]: shutdown -> are we using a single mount? or multiple mounts? also, "deletes all links"...

In the document I was proposing using "scratch/" for *hard links* that are hard for tools to distinguish. Is that still the plan?

I was proposing using a bb_clientd mount per invocation as I don't know if scratch/ allows to create directories. If it does, than a single bb_clientd mount is feasible. If it doesn't, then we must have a different bb_clientd mount per invocation id.

Comment (minor-fixes) [reply enkit/outputs/commands.go:174]: @aaahrens says you can `mkdir` in the scratch directory, so we get by with one bb_clientd instance per `output_root` directory. `shutdown` will operate on one `output_root` by unmounting bb_clientd (which removes the scratch dir and thus all the hardlinks) and deleting the convenience symlinks within

File: enkit/outputs/commands.go (hunk 1 above)
Comment (minor-fixes) [reply enkit/outputs/commands.go:177]: Agreed; I'll replace this with something less specific.

File: enkit/outputs/commands.go (hunk 2 above)
Comment (ccontavalli) [reply enkit/outputs/commands.go:174]: sgtm, then! and I assume the mounting is done dynamically, no need for shutdown.

**Review by ccontavalli — APPROVED** [review]

**Pull request merged**